src/sntp.rs

```diff
@@ -163,6 +168,22 @@ impl EspSntp {
         })
     }
 
+    pub fn subscribe<F: FnMut(core::time::Duration) + Send + 'static>(
+        &mut self,
+        callback: F,
+    ) -> EspSntpSubscription {
+        unsafe {
+            *SYNC_CB.lock().unwrap() = Some(Box::new(callback));
+            sntp_set_time_sync_notification_cb(Some(Self::sync_cb_for_subscribe));
```

Comment:
I suggest calling `sntp_set_time_sync_notification_cb` only once in the constructor of `EspSntp`, as it used to be the case.